app/controllers/assessment/autograde.rb

```diff
@@ -71,12 +74,16 @@ def regradeBatch
     submission_ids.each do |submission_id|
       submission = @assessment.submissions.find_by_id(submission_id)
       if submission
-        job = autogradeSubmissions(@course, @assessment, [submission])
-        if job == -2 # no autograding properties for this assessment
-          redirect_to([@course, @assessment, :submissions]) && return
-        elsif job < 0 # autograding failed
-          failed_jobs += 1
-          failed_list += "#{@submission.filename}: autograding error.<br>"
+        begin
+          autogradeSubmissions(@course, @assessment, [submission])
+        rescue AssessmentAutogradeHelper::AutogradeError => e
+          if e.error_code == :missing_autograding_props
+            # no autograding properties for this assessment
+            redirect_to([@course, @assessment, :submissions]) && return
+          else # autograding failed
+            failed_jobs += 1
+            failed_list += "#{@submission.filename}: autograding error.<br>"
```

Comment:
Right now this is only called in the web. I'd imagine if we add this functionality to the API we wouldn't need to be this verbose with error messages. What we could do is write a function that loops through a list of submissions, regrade them, and return a list of failed ones. This way both the API and the HTML controllers would just need to call this helper and format responses.